src/protocol/parser.rs

```diff
@@ -99,32 +99,36 @@ impl Parser {
         })
     }
 
-    pub fn parse_command(&mut self) -> R<Command> {
-        let executable = match self.parse_word()? {
-            None => self.parse_error("expected: space-separated command and arguments")?,
+    pub fn parse_command(command: &str) -> R<Command> {
```

Comment:
here's where things get opinionated-ish. This is in a separate commit so we can remove these changes if they aren't desired.

The gist of these is that we adjust `parse_command` and `parse_script_arguments` from methods that can mutate `self` into public static methods that receive a single argument. This is helpful because the callers in `protocol/mod.rs` and `protocol/command.rs` no longer need to worry about cloning input into the `original` and `input` fields.

The other stylistic change is that the `match` nested within a `loop` is refactored as a `while let Some(_)` loop, when parsing arguments into a vector of inputs.